eloquentjs/4_2_reverse_array.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Performs non-mutating array reversal operation
+ * @param {*} input 
+ * @returns 
+ */
+function reverseArray(input) {
+    if (input.constructor.toString().indexOf("Array") === -1) {
```

Comment:
Use isArray for this https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray